go.mod

```diff
@@ -0,0 +1,66 @@
+	github.com/containerd/containerd v1.2.6
+	github.com/containerd/continuity v0.0.0-20181203112020-004b46473808 // indirect
+	github.com/docker/cli v0.0.0-20181219132003-336b2a5cac7f
+	github.com/docker/distribution v2.7.1+incompatible
+	github.com/docker/docker v1.14.0-0.20190131205458-8a43b7bb99cd
+	github.com/docker/docker-credential-helpers v0.6.1 // indirect
+	github.com/docker/go-connections v0.4.0 // indirect
+	github.com/docker/go-metrics v0.0.0-20181218153428-b84716841b82 // indirect
+	github.com/docker/go-units v0.3.3 // indirect
+	github.com/docker/libtrust v0.0.0-20160708172513-aabc10ec26b7 // indirect
+	github.com/garyburd/redigo v1.6.0 // indirect
+	github.com/gofrs/uuid v3.2.0+incompatible // indirect
+	github.com/gogo/protobuf v1.2.0 // indirect
+	github.com/google/go-cmp v0.3.0 // indirect
+	github.com/gorilla/handlers v1.4.0 // indirect
+	github.com/gorilla/mux v1.7.0 // indirect
+	github.com/inconshreveable/mousetrap v1.0.0 // indirect
+	github.com/kardianos/osext v0.0.0-20170510131534-ae77be60afb1 // indirect
+	github.com/kr/pretty v0.1.0 // indirect
+	github.com/miekg/dns v0.0.0-20181005163659-0d29b283ac0f // indirect
+	github.com/morikuni/aec v0.0.0-20170113033406-39771216ff4c // indirect
+	github.com/opencontainers/go-digest v1.0.0-rc1
+	github.com/opencontainers/image-spec v1.0.1
+	github.com/opencontainers/runc v0.1.1 // indirect
+	github.com/phayes/freeport v0.0.0-20171002181615-b8543db493a5
+	github.com/pkg/errors v0.8.1
+	github.com/prometheus/client_golang v0.9.2 // indirect
+	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90 // indirect
+	github.com/prometheus/common v0.2.0 // indirect
+	github.com/prometheus/procfs v0.0.0-20190129233650-316cf8ccfec5 // indirect
+	github.com/sirupsen/logrus v1.3.0
+	github.com/spf13/cobra v0.0.3
+	github.com/spf13/pflag v1.0.3 // indirect
+	github.com/stretchr/testify v1.3.0
+	github.com/xenolf/lego v0.0.0-20160613233155-a9d8cec0e656 // indirect
+	github.com/yvasiyarov/go-metrics v0.0.0-20150112132944-c25f46c4b940 // indirect
+	github.com/yvasiyarov/gorelic v0.0.6 // indirect
+	github.com/yvasiyarov/newrelic_platform_go v0.0.0-20140908184405-b21fdbd4370f // indirect
+	golang.org/x/crypto v0.0.0-20190131182504-b8fe1690c613
+	golang.org/x/net v0.0.0-20190125091013-d26f9f9a57f3 // indirect
+	golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4 // indirect
+	golang.org/x/sys v0.0.0-20190130150945-aca44879d564 // indirect
+	golang.org/x/time v0.0.0-20181108054448-85acf8d2951c // indirect
+	google.golang.org/genproto v0.0.0-20190128161407-8ac453e89fca // indirect
+	google.golang.org/grpc v1.18.0 // indirect
+	gopkg.in/square/go-jose.v1 v1.1.2 // indirect
+	gopkg.in/yaml.v2 v2.2.2 // indirect
+	gotest.tools v2.2.0+incompatible // indirect
+	rsc.io/letsencrypt v0.0.1 // indirect
+)
+
+replace (
+	github.com/docker/docker => github.com/docker/docker v0.0.0-20190131205458-8a43b7bb99cd
+	rsc.io/letsencrypt => github.com/dmcgowan/letsencrypt v0.0.0-20160928181947-1847a81d2087
```

Comment:
Looking at the changes that were made in the fork (related to vendoring), perhaps the fork is no longer needed if I read the description of this release: https://github.com/rsc/letsencrypt/releases/tag/v0.0.1 (but probably best to make that change he upstream in the github.com/docker/distribution package first)

/cc @dmcgowan @caervs

Reply:
Trying to replace it to `replace rsc.io/letsencrypt => github.com/rsc/letsencrypt v0.0.1` breaking the tests at `github.com/deislabs/oras/pkg/auth/docker`
```
+ go test -v -covermode=atomic -coverprofile=.cover/github.com_deislabs_oras_pkg_auth_docker.cover.out github.com/deislabs/oras/pkg/auth/docker
# rsc.io/letsencrypt
../../../go/pkg/mod/github.com/rsc/letsencrypt@v0.0.1/lets.go:674:41: too many arguments in call to c.ObtainCertificate
	have ([]string, bool, nil, bool)
	want ([]string, bool, crypto.PrivateKey)
FAIL	github.com/deislabs/oras/pkg/auth/docker [build failed]
```

Reply:
Hm, interesting; https://github.com/rsc/letsencrypt/commit/33926faef6d434b854ea994228f11d0185faa0c1

looks like that may be due to a different version of https://github.com/go-acme/lego being vendored (see https://github.com/go-acme/lego/commit/01974a95cd6e66e2a9a6e49a4b9ef94ff0ae9ec6).

Don't think it has top-priority now to get rid of that fork, and it can be done upstream first (in the docker/distribution repository)

Reply:
Agree, as long as the model gives reproducibility and availability of the modules.